lib/rewards/server/subscribers/rewards-subscriber.js

```diff
@@ -0,0 +1,135 @@
+const { sc, Logger } = require('@reldens/utils');
+
+class RewardsSubscriber
+{
+    static async onBattleEnded(playerSchema, targetObject, events)
+    {
+        let rewards = sc.get(targetObject, 'rewards', false);
+
+        let winningRewards = RewardsSubscriber.getWinningRewards(rewards, false);
+        if (0 === winningRewards.length) {
+            return false;
+        }
+        let itemRepository =  targetObject.dataServer.getEntity('item')
+
+        for (let winningReward of winningRewards) {
+            let winningItem = await itemRepository.loadById(winningReward.item_id);
+            let addItemsResult = await RewardsSubscriber.addItemToPlayerInventory(playerSchema, winningItem, winningReward.drop_quantity);
+            if (!addItemsResult) {
+                Logger.error('Couldn\'t add item to Player\'s Inventory');
```

Comment:
for now try to avoid the scape quotes on the logs bucause we still need to define how those will work, just use the long way like "couldn't" > "could not" :P sorry I know is a stupid thing :P

Reply:
we should start using backticks (``), it makes string interpolation extremely easy to read.